src/models.ts

```diff
@@ -67,120 +68,202 @@ export enum SortField {
+    canSetDirection: true,
+    shownInSortBar: true,
+    shownInURL: true,
+    handledBySearchService: true,
+    searchServiceKey: 'week',
+    displayName: 'Weekly views',
+    urlNames: ['week'],
+  },
+  [SortField.title]: {
+    field: SortField.title,
+    defaultSortDirection: 'asc',
+    canSetDirection: true,
+    shownInSortBar: true,
+    shownInURL: true,
+    handledBySearchService: true,
+    searchServiceKey: 'titleSorter',
+    displayName: 'Title',
+    urlNames: ['title', 'titleSorter'],
+  },
+  [SortField.date]: {
+    field: SortField.date,
+    defaultSortDirection: 'desc',
+    canSetDirection: true,
+    shownInSortBar: true,
+    shownInURL: true,
+    handledBySearchService: true,
+    searchServiceKey: 'date',
+    displayName: 'Date published',
+    urlNames: ['date'],
+  },
+  [SortField.datearchived]: {
+    field: SortField.datearchived,
+    defaultSortDirection: 'desc',
+    canSetDirection: true,
+    shownInSortBar: true,
+    shownInURL: true,
+    handledBySearchService: true,
+    searchServiceKey: 'publicdate',
+    displayName: 'Date archived',
+    urlNames: ['publicdate'],
+  },
+  [SortField.datereviewed]: {
+    field: SortField.datereviewed,
+    defaultSortDirection: 'desc',
+    canSetDirection: true,
+    shownInSortBar: true,
+    shownInURL: true,
+    handledBySearchService: true,
+    searchServiceKey: 'reviewdate',
+    displayName: 'Date reviewed',
+    urlNames: ['reviewdate'],
+  },
+  [SortField.dateadded]: {
+    field: SortField.dateadded,
+    defaultSortDirection: 'desc',
+    canSetDirection: true,
+    shownInSortBar: true,
+    shownInURL: true,
+    handledBySearchService: true,
+    searchServiceKey: 'addeddate',
+    displayName: 'Date added',
+    urlNames: ['addeddate'],
+  },
+  [SortField.creator]: {
+    field: SortField.creator,
+    defaultSortDirection: 'asc',
+    canSetDirection: true,
+    shownInSortBar: true,
+    shownInURL: true,
+    handledBySearchService: true,
+    searchServiceKey: 'creatorSorter',
+    displayName: 'Creator',
+    urlNames: ['creator', 'creatorSorter'],
+  },
 };
 
 /**
- * Maps all allowable sort strings from the URL to their respective
- * sort fields.
+ * Returns the SortOption corresponding to the given API sort name, or
+ * the "unrecognized" SortOption if none matches.
  */
-export const URLFieldToSortField: Record<URLSortField, SortField> = {
-  week: SortField.weeklyview,
-  downloads: SortField.alltimeview,
-  title: SortField.title,
-  titleSorter: SortField.title,
-  date: SortField.date,
-  publicdate: SortField.datearchived,
-  reviewdate: SortField.datereviewed,
-  addeddate: SortField.dateadded,
-  creator: SortField.creator,
-  creatorSorter: SortField.creator,
-};
+export function sortOptionFromAPIString(sortName?: string | null): SortOption {
+  return (
+    Object.values(SORT_OPTIONS).find(opt =>
+      opt.urlNames.some(name => sortName === name)
+    ) ?? SORT_OPTIONS[SortField.unrecognized]
+  );
+}
```

Comment:
Helper function to look up the sort option that corresponds to a key from the URL or backend API.